### ISSUE TITLE
Dudas sobre API y PHP - Entrega 5

### ISSUE THREAD

**Issue by ridiaz2** [IIC2413/Syllabus-2019-2#213]
¡Hola!
Espero estén teniendo un buen cierre de semestre (:
Me surgieron algunas dudas relacionadas con el apartado **Consumir la API desde la aplicación PHP** de la Entrega 5, y agradecería mucho si pudieran ser aclaradas!
* En el primer punto aparece: "Ingresar un documento JSON a la base de datos", ¿esto hace hace referencia a la función de crear un nuevo correo? Si es así, ¿está bien que no pidamos id, ni fecha de envío, pues estos son generados automáticamente por la api?
* En el segundo punto, se pide mostrar todos los proyectos con mensajes enviados y ordenados por esta cantidad, ¿independiente del socio que haya iniciado sesión?
* En el tercer punto se pide mostrar información de socios, por lo que deberíamos guardar información sobre qué socio envió qué correo (una especie de tabla que relacione socio con correo, ya que por el momento solo tenemos información sobre el proyecto a través del cual fue enviado). ¿Esto debe ser en nuestra base de datos en codd.ing.puc.cl, o en gray.ing.puc.cl? Además, ¿solo debemos mostrar el socio con mayor cantidad de coincidencias?

Perdón por la cantidad de preguntas! No sé si es necesario separarlas en distintas *issues*, si es así, puedo cambiar a ese formato!
Muchas gracias :D 
Saludos!

**Comment by nivek0o0** [comment]
> En el primer punto aparece: "Ingresar un documento JSON a la base de datos", ¿esto hace hace referencia a la función de crear un nuevo correo? Si es así, ¿está bien que no pidamos id, ni fecha de envío, pues estos son generados automáticamente por la api?

Sí, deben ocupar la ruta para crear correos con el método POST. Está bien que no pidan lo que su aplicación es capaz de generar.

> En el segundo punto, se pide mostrar todos los proyectos con mensajes enviados y ordenados por esta cantidad, ¿independiente del socio que haya iniciado sesión?

Sí, todas las funcionalidades son independientes del tipo de inicio de sesión (deben estar en ambas). En esta funcionalidades deberían ocupar la ruta `/project-search` implementada y luego aplicar lógica de programación para obtener finalmente lo solicitado.

> En el tercer punto se pide mostrar información de socios, por lo que deberíamos guardar información sobre qué socio envió qué correo (una especie de tabla que relacione socio con correo, ya que por el momento solo tenemos información sobre el proyecto a través del cual fue enviado).

No, porque los socios se relacionan con los proyectos, entonces por transitividad puedes obtener los correos "asociados" a los socios. La funcionalidad dice _...Socio que ha enviado **a través de sus proyectos** la mayor cantidad..._

> Además, ¿solo debemos mostrar el socio con mayor cantidad de coincidencias?

Sí

**Comment by ridiaz2** [comment]
¡Perfecto! ¡Muchas gracias por lo rápido de la respuesta! Me queda súper claro.
Una duda pequeña respecto a lo de project-search solo para confirmar jaja. Esta búsqueda retorna también las coincidencias con el atributo receiver, ¿parte de esta lógica de programación sería no contar estos correos?

**Comment by nivek0o0** [comment]
Así es

**Comment by ridiaz2** [comment]
Un poco tarde jaja, pero ¡muchas graciaaas! Todo bien con la entrega :D Saludos!
